core/src/test/java/io/questdb/test/griffin/SqlOptimiserTest.java

```diff
@@ -2572,6 +2572,52 @@ public void testRangeFillWhenThereAreNoRecords() throws Exception {
         });
     }
 
+    @Test
+    public void testRewriteNegativeLimitAvoidsJoins() throws Exception {
+        assertMemoryLeak(() -> {
+            ddl("CREATE TABLE 'trades' (\n" +
+                    "  symbol SYMBOL,\n" +
+                    "  side SYMBOL,\n" +
+                    "  price DOUBLE,\n" +
+                    "  amount DOUBLE,\n" +
+                    "  timestamp TIMESTAMP\n" +
+                    ") timestamp (timestamp) PARTITION BY DAY WAL;");
+            drainWalQueue();
+            assertPlanNoLeakCheck("SELECT trades.timestamp, * FROM trades\n" +
+                    "ASOF JOIN (SELECT * from trades) trades2\n" +
+                    " LIMIT -3;", "Limit lo: -3\n" +
+                    "    SelectedRecord\n" +
+                    "        AsOf Join Fast Scan\n" +
+                    "            PageFrame\n" +
+                    "                Row forward scan\n" +
+                    "                Frame forward scan on: trades\n" +
+                    "            PageFrame\n" +
+                    "                Row forward scan\n" +
+                    "                Frame forward scan on: trades\n");
+        });
+    }
+
+    @Test
+    public void testRewriteNegativeLimitHandlesWildcards() throws Exception {
+        assertMemoryLeak(() -> {
+            ddl("CREATE TABLE 'trades' (\n" +
+                    "  symbol SYMBOL,\n" +
+                    "  side SYMBOL,\n" +
+                    "  price DOUBLE,\n" +
+                    "  amount DOUBLE,\n" +
+                    "  timestamp TIMESTAMP\n" +
+                    ") timestamp (timestamp) PARTITION BY DAY WAL;");
+            drainWalQueue();
+            assertPlanNoLeakCheck("select timestamp, * from trades limit -3", "Radix sort light\n" +
```

Comment:
Would be great to add more cases, e.g. different order `select *, timestamp` or aliases `select timestamp ts1, timestamp ts2, *` and so on.